server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -103,13 +106,16 @@ public final class IndicesRequestCache implements RemovalListener<IndicesRequest
         Property.NodeScope
     );
 
+    private final static long BASE_RAM_BYTES_USED = RamUsageEstimator.shallowSizeOfInstance(Key.class);
+
     private final ConcurrentMap<CleanupKey, Boolean> registeredClosedListeners = ConcurrentCollections.newConcurrentMap();
     private final Set<CleanupKey> keysToClean = ConcurrentCollections.newConcurrentSet();
     private final ByteSizeValue size;
     private final TimeValue expire;
     private final Cache<Key, BytesReference> cache;
+    private final IndicesService indicesService;
 
-    IndicesRequestCache(Settings settings) {
+    IndicesRequestCache(Settings settings, IndicesService indicesService) {
```

Comment:
This change makes me nervous, because this constructor is invoked from `IndicesService`'s constructor, so we're guaranteed that this instance is passed a partially-constructed instance of `IndicesService`.

Reply:
Changed this logic. Let me know what u think now. @msfroh